LIR cleanups

BruceForstall opened the pull request:
fgRemoveJTrue() and fgRemoveDeadStore() don't handle LIR form. Assert that, and
remove dead code related to embedded statements.

@pgavlin PTAL

BruceForstall committed:
LIR cleanups

fgRemoveJTrue() and fgRemoveDeadStore() don't handle LIR form. Assert that, and
remove dead code related to embedded statements.

pgavlin commented on src/jit/flowgraph.cpp:
```diff
@@ -13203,7 +13186,6 @@ bool Compiler::fgOptimizeEmptyBlock(BasicBlock* block)
                         GenTree* nop = new (this, GT_NO_OP) GenTree(GT_NO_OP, TYP_VOID);
 
-                        // TODO(pdg): allow empty blocks to be LIR blocks
                         if (compRationalIRForm)
```
Now that empty blocks can be LIR, this check should be `if (block->IsLIR())`.

pgavlin commented:
LGTM aside from one comment.

BruceForstall committed:
Change compRationalIRForm check to block->IsLIR()

BruceForstall commented:
@pgavlin -- I updated this PR by addressing your comment. There's something weird here, though, as it looks like it is considering all of your intermediate changes part of this PR. Maybe you need to give me some advanced git magic to fix this.

pgavlin commented:
LGTM
